src/oge/data_cleaning.py

```diff
@@ -292,29 +295,40 @@ def create_primary_fuel_table(gen_fuel_allocated, add_subplant_id, year):
         )
         validation.test_for_missing_subplant_id(gen_fuel_allocated, "generator_id")
 
-    # get a table of primary energy source codes by generator
-    # this will be used in `calculate_aggregated_primary_fuel()` to determine the
-    # mode of energy source codes by plant
-    # sum the fuel consumption by ESC within each generator
-    gen_primary_fuel = (
-        gen_fuel_allocated.groupby(
-            ["plant_id_eia", "subplant_id", "generator_id", "energy_source_code"],
-            dropna=False,
-        )["fuel_consumed_mmbtu"]
-        .sum()
-        .reset_index()
-    )
-
-    # only keep the ESC associated with the highest fuel consumption for each gen
-    gen_primary_fuel = gen_primary_fuel.sort_values(
-        by=["plant_id_eia", "subplant_id", "generator_id", "fuel_consumed_mmbtu"],
-        ascending=True,
-    ).drop_duplicates(
-        subset=["plant_id_eia", "subplant_id", "generator_id"], keep="last"
-    )[["plant_id_eia", "subplant_id", "generator_id", "energy_source_code"]]
-
-    if not add_subplant_id:
-        gen_primary_fuel = gen_primary_fuel.drop(columns=["subplant_id"])
+        # get a table of primary energy source codes by generator
+        # this will be used in `calculate_aggregated_primary_fuel()` to determine the
+        # mode of energy source codes by plant
+        # sum the fuel consumption by ESC within each generator
+        gen_primary_fuel = (
+            gen_fuel_allocated.groupby(
+                ["plant_id_eia", "subplant_id", "generator_id", "energy_source_code"],
+                dropna=False,
+            )["fuel_consumed_mmbtu"]
+            .sum()
+            .reset_index()
+        )
+        # only keep the ESC associated with the highest fuel consumption for each gen
+        gen_primary_fuel = gen_primary_fuel.sort_values(
+            by=["plant_id_eia", "subplant_id", "generator_id", "fuel_consumed_mmbtu"],
+            ascending=True,
+        ).drop_duplicates(
+            subset=["plant_id_eia", "subplant_id", "generator_id"], keep="last"
+        )[["plant_id_eia", "subplant_id", "generator_id", "energy_source_code"]]
+    else:
```

Comment:
Since this else block just seems to repeat lines 302-316, except removing subplant_id from the list of keys/columns, I'm wondering if we could do this more concisely by keeping the original code, but doing something like:
```
if add_subplant_id:
    keys = ["plant_id_eia","subplant_id"]
else:
    keys = ["plant_id_eia"]

gen_primary_fuel = (
            gen_fuel_allocated.groupby(
                (keys + ["generator_id", "energy_source_code"]),
                dropna=False,
            )["fuel_consumed_mmbtu"]
            .sum()
            .reset_index()
        )
```